### PR TITLE
Basic Event Editting

### LINKED ISSUE #295: Event Info should be editable
Dates/Descriptions should be editable

### PR DESCRIPTION
**Pull request by seveibar**
fixes #295

Introduces #308 and #309, but the fundamental functionality is there.

### CODE REVIEW

File: client/components/eventform/eventform.controller.js [hunk 1]
```diff
@@ -299,28 +310,48 @@ angular.module('venueApp')
...
           $scope.eventInfo = response;
           $scope.submitted = false;
+          if ($scope.onEventInfoSubmit) $scope.onEventInfoSubmit();
 
         }).catch(err => {
             err = err.data;
           });
+        }
+      }else{
+        // this is better than nothing for now
```
Comment (kmcnellis) [inline client/components/eventform/eventform.controller.js]: No its not...

Comment (seveibar) [inline client/components/eventform/eventform.controller.js]: alright I'll remove it

**Comment by seveibar** [comment]
@kmcnellis with that correction is it good to go?

**Comment by kmcnellis** [comment]
I've still got to test it

**Comment by seveibar** [comment]
thanks, I really tried to fix those two issues it introduced but I couldn't figure it out in any short period of time im afraid, i'm sure you have a better intuition

**Comment by seveibar** [comment]
@kmcnellis @leej42

**Comment by kmcnellis** [comment]
This is pretty broken still :/ You can't use the maps search bar, there are alerts instead of form validation messages everywhere, you have to upload a new image...

Also, this is event info updating, but there is no message that it will be changed for everyone. Plus, the instructor might only want to update their instructions.

**Comment by kmcnellis** [comment]
Couldn't use the same element because the google maps shows an error if its hidden

**Comment by seveibar** [comment]
Thanks Kiana, yea I think @leej42 ran into the exact same issue when he did some minor refactoring- this seems to be an error at the directive level (for that package)

File: client/components/eventform/assign/assignEvent.controller.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    function getSectionIds(){
+      window.courses = $scope.courses;
+      var sections = $scope.courses
+        .reduce((prev,course) => prev.concat(course.sections), [])
+        .filter((section)=>section.active)
+        .map((section) => section._id);
+      return sections;
+    }
+
+    $scope.submitEventAssignment = function(form){
```
Comment (seveibar) [inline client/components/eventform/assign/assignEvent.controller.js:50]: incredibly happy about the length of this function :)

Comment (seveibar) [inline client/components/eventform/assign/assignEvent.controller.js:50]: (it was previously a nightmare)

File: client/components/eventform/assign/assignEvent.controller.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        $scope.submitted = true;
+        if(getSectionIds().length == 0){
+          $scope.eventAssignmentSectionsError = true;
+          return;
+        }
+        if (form.$valid){
+          var sections = getSectionIds();
+          var createdSuccess = 0;
+          sections.forEach((sectionId)=>{
```
Comment (seveibar) [inline client/components/eventform/assign/assignEvent.controller.js:60]: This whole `createdSuccess` thing should be a `Promise.all([Promise1,Promise2....])`

File: client/components/eventform/create/createEvent.controller.js [hunk 4]
```diff
@@ -0,0 +1,327 @@
+// https://github.com/angular-ui/angular-google-maps/blob/master/example/assets/scripts/controllers/issue-624-drawing-manager.js
+'use strict';
+
+angular.module('venueApp')
```
Comment (seveibar) [inline client/components/eventform/create/createEvent.controller.js:4]: I kinda wish this controller was broken up more...

Comment (seveibar) [inline client/components/eventform/create/createEvent.controller.js:4]: I want the drawing logic separated from the form logic

File: client/components/eventform/select/selectEvent.controller.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+// https://github.com/angular-ui/angular-google-maps/blob/master/example/assets/scripts/controllers/issue-624-drawing-manager.js
```
Comment (seveibar) [inline client/components/eventform/select/selectEvent.controller.js:1]: this is a good way to break out the event info select

File: server/api/eventinfo/eventinfo.controller.js [hunk 6]
```diff
@@ -71,9 +71,7 @@ function removeEntity(res) {
 function saveEventInfoImage(files, fields, cb){
   var imagePaths = [],
       asyncTasks = [];
-  if (!files){
-    return imagePaths;
-  }
+  if (!files || !files.files){return imagePaths;}
```
Comment (seveibar) [inline server/api/eventinfo/eventinfo.controller.js:74]: if it's a single line statement, stylistically I think it should not have brackets :penguin:

File: server/api/sectionevent/sectionevent.controller.js [hunk 7]
```diff
@@ -250,9 +250,18 @@ exports.update = function(req, res) {
   if (req.body._id) {
     delete req.body._id;
   }
+  console.log("req.params.id",req.params.id);
+  console.log("req.body",req.body);
```
Comment (seveibar) [inline server/api/sectionevent/sectionevent.controller.js]: logging

File: server/config/seed.js [hunk 8]
```diff
@@ -531,7 +531,7 @@ function allSectionEvents(){
     art1Dancing: {
       section: allSections().art1._id,// Art, Community and Technology section 1
       course: allCourses().art._id,//Art, Community and Technology
-      submissionInstructions:"",
+      submissionInstructions:"Dance!",
```
Comment (seveibar) [inline server/config/seed.js:534]: points for creativity ;)

**Comment by seveibar** [comment]
The code in this PR is very clean, but I can't help but feel that drawing logic should be separated from the form logic if we're doing a refactor, I suppose it's a :penguin: but it would really be great.

The other thing is counting async completions is the exact use case for `Promise.all` so I think that should be used instead.

**Comment by kmcnellis** [comment]
I would like to do some of that as well, but this PR was getting too long and I wanted to get the functionality in.

File: server/api/eventinfo/eventinfo.controller.js (hunk 6 above)
Comment (kmcnellis) [inline server/api/eventinfo/eventinfo.controller.js:74]: That was totally part of your commit...

**Comment by kmcnellis** [comment]
Feel free to go ahead and make the simple changes. I don't have time until probably monday

**Comment by seveibar** [comment]
awesome love the promises- ill test this quickly now

**Comment by seveibar** [comment]
this creates #366 but otherwise awesome. Thanks for the fix @kmcnellis!

**Pull request merged**